src/optimization.jl

```diff
@@ -13,5 +13,6 @@ function newton_raphson(fgh!, x::AbstractArray{T}; ρ = one(T)/2, c = 1e-4, tol
         end
         x .+= search_dir*step_size
     end
-    error("Reached maximum number of iterations")
+    warn("Reached max iteration number: try increasing `max_iter` or `tol` parameters")
```

Comment:
Probably not necessary to warn before throwing an error. The `ConvergenceException` text should be reasonably informative.